src/main/java/com/atlassian/bitbucket/jenkins/internal/credentials/BitbucketCredentialsAdaptor.java

```diff
@@ -4,34 +4,47 @@
 import com.cloudbees.plugins.credentials.Credentials;
 
 import javax.annotation.Nullable;
+import javax.inject.Inject;
 
 import static com.atlassian.bitbucket.jenkins.internal.credentials.BitbucketCredentials.ANONYMOUS_CREDENTIALS;
 
-public final class BitbucketCredentialsAdaptor {
-
-    private BitbucketCredentialsAdaptor() {
-    }
-
-    public static BitbucketCredentials createWithFallback(@Nullable String credentials,
-                                                          BitbucketServerConfiguration configuration) {
-        return createWithFallback(CredentialUtils.getCredentials(credentials), configuration);
+/**
+ * For every Bitbucket instance configured on Jenkins, we have
+ * 1. Job credentials for bitbucket server which is configured by user while creating/modifying new jobs.
+ * 2. Global credentials for bitbucket server which is configured by global admin
+ *
+ * It is possible to not specify Job credentials while configuring a job. For bitbucket operation, we
+ * fall back to global configuration. This class gives the way to create bitbucket credentials based on
+ * given optional job credentials and server configuration.
+ */
+public class BitbucketCredentialsAdaptor {
```

Comment:
Personally I'm not a fan of `CredentialsUtils` because its static and can't be mocked, so maybe if we could merge it into one of the other classes that would be good?

Reply:
I have now deleted `BitbucketCredentialsAdaptor` and moved the handing to `JenkinsToBitbucketCredentialsImpl`. I can't get rid of `CredentialsUtils` since thats a bigger change.